RELEASING.md

```diff
@@ -1,89 +1,18 @@
-# How to Create a Release of OpenTelemetry Proto (for Maintainers Only)
+# How to Create a Release
 
-## Tagging the Release
+_Instruction for Maintainers only._
 
-Our release branches follow the naming convention of `v<major>.<minor>.x`, while
-the tags include the patch version `v<major>.<minor>.<patch>`. For example, the
-same branch `v0.3.x` would be used to create all `v0.3` tags (e.g. `v0.3.0`,
-`v0.3.1`).
```

Comment:
This part is still valid (even though we don't have any active release branches since we've only released straight from main and haven't published any backports so far, if we did, they'd follow this branch pattern).

Reply:
I added a paragraph to more accurately describe our current process.